### ISSUE TITLE
findResultsState does not return anything which forces a reliance on a side effect

### ISSUE THREAD

**Issue by jackwkinsey** [algolia/vue-instantsearch#765]
Currently, the `findResultsState` in `createInstantSearch.js` returns a Promise, but that promise doesn't actually resolve to anything, it merely sets the `lastResults` property of `search.helper.lastResults`. This causes a weird side effect where you have to call `getState` after `findResultsState` as awaiting `findResultsState` returns undefined.

This has caused our team some headaches when testing the integration of vue-instantsearch with our Vue application. We had to work around this side effect to ensure that we are correctly using the library.

I noticed that in the React version, `findResultsState` can be called and the results are returned allowing you to have something like `const { results } = await instantsearch.findResults(...)`.

To bring the Vue version up to par in regards to this functionality, I'd propose making the Vue version of the `findResultsState` at least return the result of calling `getState()`:

```js
return search.helper.searchOnce().then(({ content: lastResults }) => {
      // The search instance needs to act as if this was a regular `search`
      // but return a promise, since that is the interface of `asyncData`
      search.helper.lastResults = lastResults;
      // return this so promise doesn't return undefined; getState() can still be used where needed.
      return search.getState(); 
});
```

Something along those lines.

I'm mostly curious if there is a reason that the promise isn't actually resolving to anything and if it has anything to do with #755, if maybe it has something to do with the underlying helper library.

**Comment by Haroenv** [comment]
Hi @jackwkinsey, the promise isn't returning anything, since it's important that the last results are set on the instantsearch instance to be able to render correctly on the server side (with the results).

What do you think is fixed by returning the results from `findResultsState` rather than having the separate `getState`? I'd love to see what use case that would be enabling.

>  We had to work around this side effect to ensure that we are correctly using the library.

Can you clarify what you mean by that?

The fact that the old version of the helper is used here is irrelevant and fixed in the future v3 of Vue InstantSearch (#745 in particular)

**Comment by Haroenv** [comment]
In v3 this is fixed, since the result is returned from findResultsState 👍
